From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Tue, 23 Jul 2019 22:34:58 +0200
Subject: build: Automatically bump mutter API version each cycle

Mutter will now bump the API version automatically at the
beginning of a new development cycle; do the same here.

https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/641

Origin: upstream, commit:466dc8da8fa7243322fd12c8165f5a7f9d1eff29
Applied-Upstream: 3.33.90
---
 meson.build | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 15dfc43..c7ebf5e 100644
--- a/meson.build
+++ b/meson.build
@@ -4,9 +4,14 @@ project('gnome-shell', 'c',
   license: 'GPLv2+'
 )
 
+split_version = meson.project_version().split('.')
+
 # We depend on a specific version of the libmutter API. The mutter variants of
 # the Cogl and Clutter libraries also use this API version.
-mutter_api_version = '4'
+# The API version is increased automatically each development cycle,
+# starting with 0 in 3.23.x
+api_version = (split_version[1].to_int() - 23) / 2
+mutter_api_version = '@0@'.format(api_version)
 
 clutter_pc = 'mutter-clutter-' + mutter_api_version
 cogl_pc = 'mutter-cogl-' + mutter_api_version
